jetson/src/core/robot.cpp

```diff
@@ -141,22 +140,53 @@ uint16_t Robot::getDriveMotorEncoderVal(DriveMotor motor) {
 	return encoderVal[motor];
 }
 
-uint8_t Robot::getLimitVal() {
+uint8_t Robot::getLimitVals() {
+	readLimitVals();
+	return limitVals;
+}
+
+bool Robot::getLimitVal(LimitSwitch s) {
+	if (s == z_min) {
+		// z_min has no limit switch
+		return false;
+	}
 	readLimitVals();
-	return limitVal;
+	return ((limitVals & (1 << s)) != 0);
+}
+
+// ugly but required by our confusing (for this) conventions
+LimitSwitch Robot::determineLimitSwitch(ServoMotor m, bool positive) {
+	switch (m) {
+		case x:
+			if (positive) {return x_max;}
+			return x_min;
+		case y:
+			if (positive) {return y_max;}
+			return y_min;
+		default:
+			// z
+			if (positive) {return z_max;}
+			return z_min;
```

Comment:
I dont love returning z_min and then not using it later. 

Should we have some kind of fail safe? otherwise we might keep going in that direction forever? 

maybe we can have a limit on z enc position instead of limit switch and check that instead?

Reply:
made it return None instead of z_min and made the return optional